### ISSUE TITLE
QOS_XML_Handler is not wchar-compliant

### ISSUE THREAD

**Issue by mitza-oci** [objectcomputing/OpenDDS#463]
dds_qos.cpp:5702
```     
::std::basic_string< ACE_TCHAR > n (::XSCRT::XML::uq_name (e.name ()));
if (n == "sec")
```
@jwillemsen should we edit dds_qos.cpp directly or fix XSC first?

**Comment by jwillemsen** [comment]
Maybe it has been fixed in XSC, long time ago that I compiled that. If XSC can't be fixed we could directly edit dds_qos.cpp but I wouldn't prefer that

**Comment by mitza-oci** [comment]
For a short-term fix I disabled xerces when Travis CI is building the wchar configuration.  It just so happens that none of our scoreboard builds have both xerces and wchar.

Do you have time to try fixing it in XSC? I've never built that so I don't know what's required for it.

**Comment by jwillemsen** [comment]
Don't think I have much time the coming period to fix xsc, maybe just keep xerces disabled with wchar, maybe someone sponsors this in the future

**Comment by jwillemsen** [comment]
With all improvements done to the generated code this could be fixed already

**Comment by mitza-oci** [comment]
One way to find out would be to add or modify a Travis CI configuration so that both xerces and wchar are enabled.

**Comment by jwillemsen** [comment]
It does not compile but given that this is all input into IDL types which are all char maybe we should generate the xsc code with just plain char, just ignore ACE_TCHAR, what do you think @mitza-oci

**Comment by mitza-oci** [comment]
Since XSCRT is part of the ACE codebase, it's not completely unreasonable to have it use ACE_TCHAR.  It seems like the conversion really belongs to functions that were not generated by XSC, like QOS_Common.cpp.

**Comment by jwillemsen** [comment]
I have regenerated the XSC files with char, let us see how that works

**Comment by mitza-oci** [comment]
Still an issue:
```
../../../dds/DCPS/QOS_XML_Handler/QOS_DwDrTp_Base_T.cpp:191:58: error: no matching function for call to ‘QosCommon::get_qos_long(const char*)’
```

**Comment by simpsont-oci** [comment]
See #1358

**Comment by mitza-oci** [comment]
Fixed by #1358
